src/mem/alloc.h

```diff
@@ -877,16 +1084,23 @@ namespace snmalloc
 
     template<Boundary location>
     static uintptr_t
-    external_pointer(void* p, sizeclass_t sizeclass, size_t end_point)
+    external_pointer(void* p, sizeclass_t sizeclass, void* end_point)
     {
       size_t rsize = sizeclass_to_size(sizeclass);
-      size_t end_point_correction = location == End ?
-        (end_point - 1) :
-        (location == OnePastEnd ? end_point : (end_point - rsize));
-      size_t offset_from_end =
-        (end_point - 1) - static_cast<size_t>(address_cast(p));
-      size_t end_to_end = round_by_sizeclass(rsize, offset_from_end);
-      return end_point_correction - end_to_end;
+
+      void* end_point_correction = location == End ?
+        (static_cast<uint8_t*>(end_point) - 1) :
+        (location == OnePastEnd ? end_point :
+                                  (static_cast<uint8_t*>(end_point) - rsize));
```

Comment:
Should this  (and the similar cases below) not be `pointer_offset`?

Reply:
They should!  Adjusted.